x/btcstaking/docs/rewards-distribution.md

```diff
@@ -0,0 +1,145 @@
+
+The function performs the following actions:
+
+*   **Balance Check**: Verifies that the account initiating the rewards
+    distribution has a sufficient balance to cover the total rewards transfer.
+*   **Fund Transfer**: Transfers the total rewards from the sender's account to
+    the `incentive` module account. This account acts as a holding area for the
+    rewards before they are distributed.
+*   **Babylon Commission**: Calculates the Babylon commission based on the
+    commission rate defined for the BSN consumer. The commission is then sent
+    to a predefined module account.
+*   **Reward Distribution**: After deducting the Babylon commission, the
+    remaining rewards are distributed among the finality providers and their
+    corresponding BTC stakers. The distribution is based on the voting power of
+    the stakers and the commission rates of the finality providers, using the F1
+    algorithm implemented in the `x/incentive` module.
+
+## Distribution via `MsgAddBsnRewards` Message
+
+The message used to trigger rewards distribution is defined as follows:
+
+```protobuf
+type MsgAddBsnRewards struct {
+	// Sender is the babylon address which will pay for the rewards
+	Sender string `protobuf:"bytes,1,opt,name=sender,proto3" json:"sender,omitempty"`
+	// BsnConsumerId is the ID of the BSN consumer
+	// - for Cosmos SDK chains, the consumer ID will be the IBC client ID
+	// - for rollup chains, the consumer ID will be the chain ID of the rollup
+	//   chain
+	BsnConsumerId string `protobuf:"bytes,2,opt,name=bsn_consumer_id,json=bsnConsumerId,proto3" json:"bsn_consumer_id,omitempty"`
+	// TotalRewards is the total amount of rewards to be distributed among finality providers.
+	// This amount will be distributed according to the ratios specified in fp_ratios.
+	TotalRewards github_com_cosmos_cosmos_sdk_types.Coins `protobuf:"bytes,3,rep,name=total_rewards,json=totalRewards,proto3,castrepeated=github.com/cosmos/cosmos-sdk/types.Coins" json:"total_rewards"`
+	// FpRatios is a list of finality providers and their respective reward distribution ratios.
+	// The ratios should sum to 1.0 to distribute the entire total_rewards amount.
+	FpRatios []FpRatio `protobuf:"bytes,4,rep,name=fp_ratios,json=fpRatios,proto3" json:"fp_ratios"`
+}
+```
+
+> **⚡ Important** Before sending the message, the `Sender` must have enough
+> coins to cover the amount declared in the `TotalRewards` field.
+
+> **⚡ Important** All finality providers in the `FpRatios` list must already be
+> registered on the Babylon chain and have active delegations. Otherwise, an
+> error will be returned to the caller.
+
+> **⚡ Important** The consumer identified by `BsnConsumerId` must exist on
+> Babylon Genesis.
+
+When Babylon Genesis receives a `MsgAddBsnRewards` message, it extracts all
+relevant fields and delegates processing to the `AddBsnRewards` function
+described earlier.
+
+> **⚡ Important** Babylon does not enforce how the `FpRatios` must be
```

Comment:
Might be worth to mention that the ratios must sum to 1, even though is already noted in the `FpRatios` attribute in `MsgAddBsnRewards struct`